### PR TITLE
Add pip caching and correct random seed in tests

### PR DESCRIPTION
**Pull request by mmuckley**
Add pip caching to speed up CI. Also corrects a random seed issue for tests in CI.

### CODE REVIEW

File: tests/test_distortion.py [hunk 1]
```diff
@@ -26,6 +27,12 @@ def tf_ms_ssim(x, y):
     return torch.tensor(tf.image.ssim_multiscale(x, y, max_val=1).numpy())
 
 
+def rand_im(shape, seed=12345):
```
Comment (jordan-benjamin) [inline tests/test_distortion.py:30]: Wouldn't it be easier to just use `torch.manual_seed(seed)` at the top of each test and just keep the original code, therefore you wouldn't have to pass a different seed every time you created a new tensor?

Alternatively, we could add a `setup_function` in the file that automatically sets the seeds before each tests, as per [the pytest docs](https://docs.pytest.org/en/6.2.x/xunit_setup.html).

File: tests/test_distortion.py [hunk 2]
```diff
@@ -70,11 +77,11 @@ def test_ms_ssim_module(num_channels, img_size, batch_sizes):
     metric = MultiscaleStructuralSimilarity()
 
     imgs1 = [
-        torch.rand([batch_size, num_channels, img_size, img_size])
+        rand_im([batch_size, num_channels, img_size, img_size], 123)
```
Comment (jordan-benjamin) [inline tests/test_distortion.py:80]: was the intention here to use the same seed for every tensor in the list? I don't think this matters based on the specific inputs sizes we have, but if `batch_sizes` has duplicate entries then duplicate tensors will show up here.

File: tests/test_distortion.py [hunk 3]
```diff
@@ -106,8 +113,8 @@ def test_lpips(batch_sizes, input_shape):
 
     metric = LearnedPerceptualImagePatchSimilarity()
 
-    imgs1 = [torch.rand([batch_size, 3, *input_shape]) for batch_size in batch_sizes]
-    imgs2 = [torch.rand([batch_size, 3, *input_shape]) for batch_size in batch_sizes]
+    imgs1 = [rand_im([batch_size, 3, *input_shape], 0) for batch_size in batch_sizes]
```
Comment (jordan-benjamin) [inline tests/test_distortion.py:116]: same note about the lists as above

**Review by jordan-benjamin — APPROVED** [review]
LGTM! Left a few comments that might clean things up.

File: tests/test_distortion.py (hunk 1 above)
Comment (mmuckley) [reply tests/test_distortion.py:30]: I don't think so. That will change the global seed. The seed should be local to each test.

File: tests/test_distortion.py (hunk 2 above)
Comment (mmuckley) [reply tests/test_distortion.py:80]: Nice catch - this is a bug.
